pkg/storage/storage.go

```diff
@@ -668,3 +679,70 @@ func (s *storageModule) Monitor(ctx context.Context) <-chan pkg.PoolsStats {
 
 	return ch
 }
+
+func (s *storageModule) periodicallyCheckDiskShutdown() {
+	ticker := time.NewTicker(5 * time.Minute)
+	quit := make(chan struct{})
```

Comment:
You aware this channel is never getting any input? the write way to do it is to pass a `Context` object to this function, then the caller can call the cancel method.

Reply:
If you don't need a context (or don't want one) you can simply drop the select stmt and instead have a loop like

```go
for {
 <- ticker.C
 s.shutdownDisks()
}
```